phac-nml/irida-next
# Uploader policy update (DFCT0010069)

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR updates policy methods for a user with uploader access level to be able to create a sample, modify a sample (create/remove files, update sample metadata), and read projects

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/phac-nml/irida-next/assets/25466211/288ade76-9c91-49d1-92b9-d7989f38f082)

![image](https://github.com/phac-nml/irida-next/assets/25466211/e2012d02-2443-4ce1-954d-6024de8c029b)


## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Start up irida next
2. Login as any user
3. Create a project and a group
4. Add bot accounts to both with an `UPLOADER` access level. Copy down the tokens
5. Generate new tokens for both bot accounts and set the Expiration to an earlier date. Copy down the token
6. Use `Postman` to test out the graphql queries
7. In Postman click the icon/button next to `Untitled Request` and select `Graphql`
8. In the `URL` input, type in `http://localhost:3000/api/graphql`
9. Click on the `authorization` tab. Here you can run the query/mutation with the bot accounts using their email and the tokens as the password. Make sure to try out all the queries/mutations using the valid and expired tokens for each bot account. Verify the bot account with the `UPLOADER` access level can query/mutate if they have a valid token, otherwise they should not be able to.
10. Take a look in the `test/graphql/` directory for all the queries and mutations and to see how the data is passed in to each query and mutation. 

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
